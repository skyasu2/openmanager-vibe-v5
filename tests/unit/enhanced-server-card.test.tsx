import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import EnhancedServerCard from '@/components/dashboard/EnhancedServerCard';
import { vi } from 'vitest';

// 💡 스냅샷 테스트용 목업 서버 데이터
const mockServer = {
  id: 'api-eu-045',
  hostname: 'api-eu-045.internal',
  name: 'api-eu-045',
  type: 'api',
  environment: 'production',
  location: 'EU West',
  provider: 'AWS',
  status: 'warning' as const,
  cpu: 67,
  memory: 72,
  disk: 54,
  network: 45,
  uptime: '8일 12시간',
  lastUpdate: new Date(),
  alerts: 2,
  services: [
    { name: 'nodejs', status: 'running' as const, port: 3000 },
    { name: 'nginx', status: 'running' as const, port: 80 },
  ],
  specs: {
    cpu_cores: 4,
    memory_gb: 16,
    disk_gb: 100,
  },
  os: 'Ubuntu 22.04',
  ip: '10.0.0.12',
  networkStatus: 'good' as const,
};

describe('EnhancedServerCard', () => {
  it('renders correctly and matches snapshot', () => {
    // 🔒 Math.random 및 Date.now 고정으로 스냅샷 결정론적 생성
    const originalRandom = Math.random;
    const originalNow = Date.now;
    Math.random = () => 0.42; // 고정된 임의 값
    Date.now = () => 1718800000000; // 2025-06-20T00:00:00.000Z

    const { container } = render(
      <EnhancedServerCard server={mockServer as any} index={0} />
    );

    expect(container).toMatchSnapshot();

    // 복원
    Math.random = originalRandom;
    Date.now = originalNow;

    // 모듈 캐시 초기화로 이후 테스트 영향 제거
    vi.resetModules();
  });
});
