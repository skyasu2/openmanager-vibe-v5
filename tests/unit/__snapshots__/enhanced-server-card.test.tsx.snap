// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EnhancedServerCard > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="
        relative p-6 min-h-[240px] rounded-xl cursor-pointer
        bg-gradient-to-br from-yellow-50 via-amber-50 to-orange-50
        border-2 border-yellow-200 hover:border-yellow-300
        shadow-lg shadow-yellow-100 hover:shadow-xl
        transition-all duration-300 ease-out
        backdrop-blur-sm
        group
      "
    style="opacity: 0; transform: translateY(20px) scale(0.9);"
  >
    <div
      class="absolute top-3 right-3 flex items-center gap-2"
    >
      <div
        class="w-2 h-2 bg-green-400 rounded-full"
      />
      <svg
        aria-hidden="true"
        class="lucide lucide-minus w-3 h-3 text-gray-400"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12h14"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="lucide lucide-wifi w-4 h-4 text-blue-500"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 20h.01"
        />
        <path
          d="M2 8.82a15 15 0 0 1 20 0"
        />
        <path
          d="M5 12.859a10 10 0 0 1 14 0"
        />
        <path
          d="M8.5 16.429a5 5 0 0 1 7 0"
        />
      </svg>
    </div>
    <div
      class="flex items-start justify-between mb-4"
    >
      <div
        class="flex items-center gap-3"
      >
        <div
          class="p-2 rounded-lg bg-yellow-100 text-yellow-600"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-git-branch w-5 h-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="6"
              x2="6"
              y1="3"
              y2="15"
            />
            <circle
              cx="18"
              cy="6"
              r="3"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <path
              d="M18 9a9 9 0 0 1-9 9"
            />
          </svg>
        </div>
        <div>
          <h3
            class="font-bold text-gray-900 text-lg group-hover:text-gray-700 transition-colors flex items-center gap-2"
          >
            <span>
              api-eu-045
            </span>
          </h3>
          <p
            class="text-sm text-gray-600"
          >
            api
             • 
            EU West
          </p>
        </div>
      </div>
      <div
        class="px-3 py-1 rounded-full bg-yellow-100 flex items-center gap-1"
      >
        <span>
          ⚠️
        </span>
        <span
          class="text-xs font-semibold text-yellow-800"
        >
          경고
        </span>
      </div>
    </div>
    <div
      class="space-y-4"
    >
      <div
        class="grid grid-cols-4 gap-3 bg-white/70 rounded-lg p-4 backdrop-blur-sm"
      >
        <div
          class="flex flex-col items-center group"
        >
          <div
            class="flex items-center gap-1 mb-2"
          >
            <div
              class="text-gray-500 group-hover:scale-110 transition-transform"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-cpu w-3 h-3"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 20v2"
                />
                <path
                  d="M12 2v2"
                />
                <path
                  d="M17 20v2"
                />
                <path
                  d="M17 2v2"
                />
                <path
                  d="M2 12h2"
                />
                <path
                  d="M2 17h2"
                />
                <path
                  d="M2 7h2"
                />
                <path
                  d="M20 12h2"
                />
                <path
                  d="M20 17h2"
                />
                <path
                  d="M20 7h2"
                />
                <path
                  d="M7 20v2"
                />
                <path
                  d="M7 2v2"
                />
                <rect
                  height="16"
                  rx="2"
                  width="16"
                  x="4"
                  y="4"
                />
                <rect
                  height="8"
                  rx="1"
                  width="8"
                  x="8"
                  y="8"
                />
              </svg>
            </div>
            <span
              class="text-xs font-medium text-gray-700"
            >
              CPU
            </span>
          </div>
          <div
            class="w-20 h-12 relative bg-white/80 rounded-lg p-1 shadow-sm"
          >
            <svg
              class="w-full h-full"
              preserveAspectRatio="none"
              viewBox="0 0 100 100"
            >
              <defs>
                <lineargradient
                  id="gradient-api-eu-045-CPU-0.34290702109165627"
                  x1="0%"
                  x2="0%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    stop-color="#ef4444"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="50%"
                    stop-color="#ef4444"
                    stop-opacity="0.4"
                  />
                  <stop
                    offset="100%"
                    stop-color="#ef4444"
                    stop-opacity="0.1"
                  />
                </lineargradient>
                <filter
                  id="glow-api-eu-045-CPU-0.22175199671238288"
                >
                  <fegaussianblur
                    result="coloredBlur"
                    stdDeviation="2"
                  />
                  <femerge>
                    <femergenode
                      in="coloredBlur"
                    />
                    <femergenode
                      in="SourceGraphic"
                    />
                  </femerge>
                </filter>
              </defs>
              <defs>
                <pattern
                  height="10"
                  id="grid-api-eu-045-CPU"
                  patternUnits="userSpaceOnUse"
                  width="10"
                >
                  <path
                    d="M 10 0 L 0 0 0 10"
                    fill="none"
                    stroke="#e2e8f0"
                    stroke-width="0.3"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#grid-api-eu-045-CPU)"
                height="100"
                opacity="0.3"
                width="100"
              />
              <polygon
                class="transition-all duration-300"
                fill="url(#gradient-api-eu-045-CPU-0.34290702109165627)"
                points="0,100 0,38.609953950253 9.090909090909092,33.865901237845705 18.181818181818183,36.914500016615946 27.27272727272727,35.33538762542918 36.36363636363637,21.09372839877898 45.45454545454545,36.50582628110216 54.54545454545454,29.472311854512128 63.63636363636363,18.766239089786367 72.72727272727273,28.45207229948683 81.81818181818183,30.387840310804023 90.9090909090909,31.19672316454478 100,40.6175338425186 100,100"
              />
              <polyline
                class="transition-all duration-300"
                fill="none"
                filter="url(#glow-api-eu-045-CPU-0.22175199671238288)"
                points="0,38.609953950253 9.090909090909092,33.865901237845705 18.181818181818183,36.914500016615946 27.27272727272727,35.33538762542918 36.36363636363637,21.09372839877898 45.45454545454545,36.50582628110216 54.54545454545454,29.472311854512128 63.63636363636363,18.766239089786367 72.72727272727273,28.45207229948683 81.81818181818183,30.387840310804023 90.9090909090909,31.19672316454478 100,40.6175338425186"
                stroke="#ef4444"
                stroke-width="2.5"
                vector-effect="non-scaling-stroke"
              />
              <circle
                cx="100"
                cy="40.6175338425186"
                fill="#ef4444"
                filter="url(#glow-api-eu-045-CPU-0.22175199671238288)"
                r="2.5"
                stroke="white"
                stroke-width="1.5"
              />
            </svg>
          </div>
          <div
            class="text-sm font-bold mt-1 px-2 py-1 rounded-full bg-white/80"
            style="color: rgb(239, 68, 68);"
          >
            59
            %
          </div>
        </div>
        <div
          class="flex flex-col items-center group"
        >
          <div
            class="flex items-center gap-1 mb-2"
          >
            <div
              class="text-gray-500 group-hover:scale-110 transition-transform"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-activity w-3 h-3"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 12h-2.48a2 2 0 0 0-1.93 1.46l-2.35 8.36a.25.25 0 0 1-.48 0L9.24 2.18a.25.25 0 0 0-.48 0l-2.35 8.36A2 2 0 0 1 4.49 12H2"
                />
              </svg>
            </div>
            <span
              class="text-xs font-medium text-gray-700"
            >
              메모리
            </span>
          </div>
          <div
            class="w-20 h-12 relative bg-white/80 rounded-lg p-1 shadow-sm"
          >
            <svg
              class="w-full h-full"
              preserveAspectRatio="none"
              viewBox="0 0 100 100"
            >
              <defs>
                <lineargradient
                  id="gradient-api-eu-045-메모리-0.6586378549986167"
                  x1="0%"
                  x2="0%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    stop-color="#3b82f6"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="50%"
                    stop-color="#3b82f6"
                    stop-opacity="0.4"
                  />
                  <stop
                    offset="100%"
                    stop-color="#3b82f6"
                    stop-opacity="0.1"
                  />
                </lineargradient>
                <filter
                  id="glow-api-eu-045-메모리-0.34669480128006014"
                >
                  <fegaussianblur
                    result="coloredBlur"
                    stdDeviation="2"
                  />
                  <femerge>
                    <femergenode
                      in="coloredBlur"
                    />
                    <femergenode
                      in="SourceGraphic"
                    />
                  </femerge>
                </filter>
              </defs>
              <defs>
                <pattern
                  height="10"
                  id="grid-api-eu-045-메모리"
                  patternUnits="userSpaceOnUse"
                  width="10"
                >
                  <path
                    d="M 10 0 L 0 0 0 10"
                    fill="none"
                    stroke="#e2e8f0"
                    stroke-width="0.3"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#grid-api-eu-045-메모리)"
                height="100"
                opacity="0.3"
                width="100"
              />
              <polygon
                class="transition-all duration-300"
                fill="url(#gradient-api-eu-045-메모리-0.6586378549986167)"
                points="0,100 0,35.86815722627003 9.090909090909092,25.897983199752517 18.181818181818183,27.961225969556892 27.27272727272727,19.258039572375083 36.36363636363637,29.064729413282336 45.45454545454545,29.685609893524003 54.54545454545454,36.78974416008418 63.63636363636363,32.19542983767157 72.72727272727273,25.100158034018136 81.81818181818183,35.6694155994133 90.9090909090909,24.689772432864572 100,20.504247646124753 100,100"
              />
              <polyline
                class="transition-all duration-300"
                fill="none"
                filter="url(#glow-api-eu-045-메모리-0.34669480128006014)"
                points="0,35.86815722627003 9.090909090909092,25.897983199752517 18.181818181818183,27.961225969556892 27.27272727272727,19.258039572375083 36.36363636363637,29.064729413282336 45.45454545454545,29.685609893524003 54.54545454545454,36.78974416008418 63.63636363636363,32.19542983767157 72.72727272727273,25.100158034018136 81.81818181818183,35.6694155994133 90.9090909090909,24.689772432864572 100,20.504247646124753"
                stroke="#3b82f6"
                stroke-width="2.5"
                vector-effect="non-scaling-stroke"
              />
              <circle
                cx="100"
                cy="20.504247646124753"
                fill="#3b82f6"
                filter="url(#glow-api-eu-045-메모리-0.34669480128006014)"
                r="2.5"
                stroke="white"
                stroke-width="1.5"
              />
            </svg>
          </div>
          <div
            class="text-sm font-bold mt-1 px-2 py-1 rounded-full bg-white/80"
            style="color: rgb(59, 130, 246);"
          >
            79
            %
          </div>
        </div>
        <div
          class="flex flex-col items-center group"
        >
          <div
            class="flex items-center gap-1 mb-2"
          >
            <div
              class="text-gray-500 group-hover:scale-110 transition-transform"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-hard-drive w-3 h-3"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="22"
                  x2="2"
                  y1="12"
                  y2="12"
                />
                <path
                  d="M5.45 5.11 2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
                />
                <line
                  x1="6"
                  x2="6.01"
                  y1="16"
                  y2="16"
                />
                <line
                  x1="10"
                  x2="10.01"
                  y1="16"
                  y2="16"
                />
              </svg>
            </div>
            <span
              class="text-xs font-medium text-gray-700"
            >
              디스크
            </span>
          </div>
          <div
            class="w-20 h-12 relative bg-white/80 rounded-lg p-1 shadow-sm"
          >
            <svg
              class="w-full h-full"
              preserveAspectRatio="none"
              viewBox="0 0 100 100"
            >
              <defs>
                <lineargradient
                  id="gradient-api-eu-045-디스크-0.41704447317093996"
                  x1="0%"
                  x2="0%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    stop-color="#8b5cf6"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="50%"
                    stop-color="#8b5cf6"
                    stop-opacity="0.4"
                  />
                  <stop
                    offset="100%"
                    stop-color="#8b5cf6"
                    stop-opacity="0.1"
                  />
                </lineargradient>
                <filter
                  id="glow-api-eu-045-디스크-0.776442284127159"
                >
                  <fegaussianblur
                    result="coloredBlur"
                    stdDeviation="2"
                  />
                  <femerge>
                    <femergenode
                      in="coloredBlur"
                    />
                    <femergenode
                      in="SourceGraphic"
                    />
                  </femerge>
                </filter>
              </defs>
              <defs>
                <pattern
                  height="10"
                  id="grid-api-eu-045-디스크"
                  patternUnits="userSpaceOnUse"
                  width="10"
                >
                  <path
                    d="M 10 0 L 0 0 0 10"
                    fill="none"
                    stroke="#e2e8f0"
                    stroke-width="0.3"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#grid-api-eu-045-디스크)"
                height="100"
                opacity="0.3"
                width="100"
              />
              <polygon
                class="transition-all duration-300"
                fill="url(#gradient-api-eu-045-디스크-0.41704447317093996)"
                points="0,100 0,41.4235094761561 9.090909090909092,46.05213459749653 18.181818181818183,45.60598706463597 27.27272727272727,49.52220107538916 36.36363636363637,47.09438856360374 45.45454545454545,49.50229933639182 54.54545454545454,47.997642473126675 63.63636363636363,48.16485126818385 72.72727272727273,46.0440114988353 81.81818181818183,48.634578893086854 90.9090909090909,44.17583450633847 100,45.76360430850656 100,100"
              />
              <polyline
                class="transition-all duration-300"
                fill="none"
                filter="url(#glow-api-eu-045-디스크-0.776442284127159)"
                points="0,41.4235094761561 9.090909090909092,46.05213459749653 18.181818181818183,45.60598706463597 27.27272727272727,49.52220107538916 36.36363636363637,47.09438856360374 45.45454545454545,49.50229933639182 54.54545454545454,47.997642473126675 63.63636363636363,48.16485126818385 72.72727272727273,46.0440114988353 81.81818181818183,48.634578893086854 90.9090909090909,44.17583450633847 100,45.76360430850656"
                stroke="#8b5cf6"
                stroke-width="2.5"
                vector-effect="non-scaling-stroke"
              />
              <circle
                cx="100"
                cy="45.76360430850656"
                fill="#8b5cf6"
                filter="url(#glow-api-eu-045-디스크-0.776442284127159)"
                r="2.5"
                stroke="white"
                stroke-width="1.5"
              />
            </svg>
          </div>
          <div
            class="text-sm font-bold mt-1 px-2 py-1 rounded-full bg-white/80"
            style="color: rgb(139, 92, 246);"
          >
            54
            %
          </div>
        </div>
        <div
          class="flex flex-col items-center group"
        >
          <div
            class="flex items-center gap-1 mb-2"
          >
            <div
              class="text-gray-500 group-hover:scale-110 transition-transform"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-network w-3 h-3"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="6"
                  rx="1"
                  width="6"
                  x="16"
                  y="16"
                />
                <rect
                  height="6"
                  rx="1"
                  width="6"
                  x="2"
                  y="16"
                />
                <rect
                  height="6"
                  rx="1"
                  width="6"
                  x="9"
                  y="2"
                />
                <path
                  d="M5 16v-3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3"
                />
                <path
                  d="M12 12V8"
                />
              </svg>
            </div>
            <span
              class="text-xs font-medium text-gray-700"
            >
              네트워크
            </span>
          </div>
          <div
            class="w-20 h-12 relative bg-white/80 rounded-lg p-1 shadow-sm"
          >
            <svg
              class="w-full h-full"
              preserveAspectRatio="none"
              viewBox="0 0 100 100"
            >
              <defs>
                <lineargradient
                  id="gradient-api-eu-045-네트워크-0.10235985657267221"
                  x1="0%"
                  x2="0%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    stop-color="#10b981"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="50%"
                    stop-color="#10b981"
                    stop-opacity="0.4"
                  />
                  <stop
                    offset="100%"
                    stop-color="#10b981"
                    stop-opacity="0.1"
                  />
                </lineargradient>
                <filter
                  id="glow-api-eu-045-네트워크-0.2396508313995953"
                >
                  <fegaussianblur
                    result="coloredBlur"
                    stdDeviation="2"
                  />
                  <femerge>
                    <femergenode
                      in="coloredBlur"
                    />
                    <femergenode
                      in="SourceGraphic"
                    />
                  </femerge>
                </filter>
              </defs>
              <defs>
                <pattern
                  height="10"
                  id="grid-api-eu-045-네트워크"
                  patternUnits="userSpaceOnUse"
                  width="10"
                >
                  <path
                    d="M 10 0 L 0 0 0 10"
                    fill="none"
                    stroke="#e2e8f0"
                    stroke-width="0.3"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#grid-api-eu-045-네트워크)"
                height="100"
                opacity="0.3"
                width="100"
              />
              <polygon
                class="transition-all duration-300"
                fill="url(#gradient-api-eu-045-네트워크-0.10235985657267221)"
                points="0,100 0,52.246586212367276 9.090909090909092,49.087893026562085 18.181818181818183,50.645200319729256 27.27272727272727,47.6000615706601 36.36363636363637,51.25851326731228 45.45454545454545,54.096089649908265 54.54545454545454,35.0534156417027 63.63636363636363,74.99712716060306 72.72727272727273,65.8623753889954 81.81818181818183,51.656233387052154 90.9090909090909,65.22870746292757 100,35.33825236876534 100,100"
              />
              <polyline
                class="transition-all duration-300"
                fill="none"
                filter="url(#glow-api-eu-045-네트워크-0.2396508313995953)"
                points="0,52.246586212367276 9.090909090909092,49.087893026562085 18.181818181818183,50.645200319729256 27.27272727272727,47.6000615706601 36.36363636363637,51.25851326731228 45.45454545454545,54.096089649908265 54.54545454545454,35.0534156417027 63.63636363636363,74.99712716060306 72.72727272727273,65.8623753889954 81.81818181818183,51.656233387052154 90.9090909090909,65.22870746292757 100,35.33825236876534"
                stroke="#10b981"
                stroke-width="2.5"
                vector-effect="non-scaling-stroke"
              />
              <circle
                cx="100"
                cy="35.33825236876534"
                fill="#10b981"
                filter="url(#glow-api-eu-045-네트워크-0.2396508313995953)"
                r="2.5"
                stroke="white"
                stroke-width="1.5"
              />
            </svg>
          </div>
          <div
            class="text-sm font-bold mt-1 px-2 py-1 rounded-full bg-white/80"
            style="color: rgb(16, 185, 129);"
          >
            65
            %
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-4 gap-2"
      >
        <div
          class="text-center"
        >
          <div
            class="text-xs text-gray-600 flex items-center justify-center gap-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-cpu w-3 h-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 20v2"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M17 20v2"
              />
              <path
                d="M17 2v2"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M2 17h2"
              />
              <path
                d="M2 7h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="M20 17h2"
              />
              <path
                d="M20 7h2"
              />
              <path
                d="M7 20v2"
              />
              <path
                d="M7 2v2"
              />
              <rect
                height="16"
                rx="2"
                width="16"
                x="4"
                y="4"
              />
              <rect
                height="8"
                rx="1"
                width="8"
                x="8"
                y="8"
              />
            </svg>
            CPU
          </div>
          <div
            class="text-sm font-bold text-red-600"
          >
            67
            %
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="text-xs text-gray-600 flex items-center justify-center gap-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-activity w-3 h-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 12h-2.48a2 2 0 0 0-1.93 1.46l-2.35 8.36a.25.25 0 0 1-.48 0L9.24 2.18a.25.25 0 0 0-.48 0l-2.35 8.36A2 2 0 0 1 4.49 12H2"
              />
            </svg>
            RAM
          </div>
          <div
            class="text-sm font-bold text-blue-600"
          >
            72
            %
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="text-xs text-gray-600 flex items-center justify-center gap-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-hard-drive w-3 h-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="22"
                x2="2"
                y1="12"
                y2="12"
              />
              <path
                d="M5.45 5.11 2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
              />
              <line
                x1="6"
                x2="6.01"
                y1="16"
                y2="16"
              />
              <line
                x1="10"
                x2="10.01"
                y1="16"
                y2="16"
              />
            </svg>
            디스크
          </div>
          <div
            class="text-sm font-bold text-purple-600"
          >
            54
            %
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="text-xs text-gray-600 flex items-center justify-center gap-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-network w-3 h-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="6"
                rx="1"
                width="6"
                x="16"
                y="16"
              />
              <rect
                height="6"
                rx="1"
                width="6"
                x="2"
                y="16"
              />
              <rect
                height="6"
                rx="1"
                width="6"
                x="9"
                y="2"
              />
              <path
                d="M5 16v-3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3"
              />
              <path
                d="M12 12V8"
              />
            </svg>
            네트워크
          </div>
          <div
            class="text-sm font-bold text-green-600"
          >
            45
            %
          </div>
        </div>
      </div>
      <div
        class="flex gap-2 flex-wrap"
      >
        <div
          class="px-2 py-1 rounded text-xs flex items-center gap-1 bg-green-100 text-green-700"
        >
          <div
            class="w-1.5 h-1.5 rounded-full bg-green-500"
          />
          nodejs
        </div>
        <div
          class="px-2 py-1 rounded text-xs flex items-center gap-1 bg-green-100 text-green-700"
        >
          <div
            class="w-1.5 h-1.5 rounded-full bg-green-500"
          />
          nginx
        </div>
      </div>
      <div
        class="flex items-center justify-between p-2 bg-white/60 rounded-lg"
      >
        <div
          class="flex items-center gap-2"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-wifi w-4 h-4 text-blue-500"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 20h.01"
            />
            <path
              d="M2 8.82a15 15 0 0 1 20 0"
            />
            <path
              d="M5 12.859a10 10 0 0 1 14 0"
            />
            <path
              d="M8.5 16.429a5 5 0 0 1 7 0"
            />
          </svg>
          <span
            class="text-xs font-medium text-gray-700"
          >
            네트워크 상태
          </span>
        </div>
        <span
          class="text-xs font-semibold px-2 py-1 rounded-full bg-blue-100 text-blue-700"
        >
          양호
        </span>
      </div>
      <div
        class="flex items-center gap-2 p-2 bg-red-50 text-red-700 rounded-lg text-sm"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-triangle-alert w-4 h-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3"
          />
          <path
            d="M12 9v4"
          />
          <path
            d="M12 17h.01"
          />
        </svg>
        <span
          class="font-medium"
        >
          2
          개 알림
        </span>
      </div>
    </div>
  </div>
</div>
`;
